### PR TITLE
Pu/destructors

### PR DESCRIPTION
**Pull request by izulin**
### Context

Introduces destructors for basic components of the project.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #775 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [ ] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ module.exports = {
     "<rootDir>/test/**/*spec.(ts|js)"
   ],
 
-  silent: true,
+  silent: false,
```
Comment (izulin) [inline jest.config.js:47]:
```suggestion
  silent: true,
```

**Comment by wojciechczerniak** [comment]
To much of complexity. JavaScript should handle destruction just fine in GC cyle. It is faster than #788 but IMO not worth it. Final result of cleanup is exactly the same. I propose to close this PR and focus on #788

**Comment by izulin** [comment]
> To much of complexity. JavaScript should handle destruction just fine in GC cyle. It is faster than #788 but IMO not worth it. Final result of cleanup is exactly the same. I propose to close this PR and focus on #788

Ok!

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
To much of complexity. JavaScript should handle destruction just fine in GC cyle. It is faster than #788 but IMO not worth it. Final result of cleanup is exactly the same. I propose to close this PR and focus on #788

**Pull request closed without merging**